arrow-test/src/lib.rs

```diff
@@ -72,6 +78,70 @@ impl ScalarUDFImpl for CidString {
     }
 }
 
+/// ScalarUDF to convert a binary CID into a string for easier inspection.
+#[derive(Debug)]
+pub struct CidStringList {
+    signature: Signature,
+}
+
+impl Default for CidStringList {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CidStringList {
+    /// Construct new instance
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::new(
+                TypeSignature::Exact(vec![DataType::new_list(DataType::Binary, true)]),
+                Volatility::Immutable,
+            ),
+        }
+    }
+}
+
+impl ScalarUDFImpl for CidStringList {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+    fn name(&self) -> &str {
+        "array_cid_string"
+    }
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+    fn return_type(&self, _args: &[DataType]) -> datafusion::common::Result<DataType> {
+        Ok(DataType::new_list(DataType::Utf8, true))
+    }
+    fn invoke(&self, args: &[ColumnarValue]) -> datafusion::common::Result<ColumnarValue> {
+        let args = ColumnarValue::values_to_arrays(args)?;
+        let all_cids = as_list_array(&args[0])?;
+        let mut strs = ListBuilder::new(StringBuilder::new());
+        for cids in ArrayIter::new(all_cids) {
+            if let Some(cids) = cids {
```

Comment:
I find this odd (that we have values that are None) but I guess it's expected based on the Nullable definition. Does the DataType need to be nullable (and when would it be in practice)? I know this for testing so it's not a big deal, just trying to understand for when I have to add something.

Reply:
yeah not real use case for when a cid would be null. For example the event_cid column is not nullable. However a this is a function that can be used on any function its best to not assume data is not null. And just pass nulls along.